flytestdlib/cache/auto_refresh.go

```diff
@@ -406,3 +428,8 @@ func NewAutoRefreshCache(name string, syncCb SyncFunc, syncRateLimiter workqueue
 
 	return NewAutoRefreshBatchedCache(name, SingleItemBatches, syncCb, syncRateLimiter, resyncPeriod, parallelizm, size, scope)
 }
+
+func newAutoRefreshCacheWithClock(name string, syncCb SyncFunc, syncRateLimiter workqueue.RateLimiter, resyncPeriod time.Duration,
```

Comment:
I've added these package lever constructors for testing but honestly with all the arguments that can be configured on the auto refresh cache it would be nicer to have a constructor with `Options`. I can follow up with that in another pull request if you'd like.

Reply:
We can do that in a separate PR.